app/controllers/icons_controller.rb

```diff
@@ -49,6 +49,20 @@ def show
       @posts = posts_from_relation(Post.where(where_calc).ordered)
     elsif params[:view] == 'galleries'
       use_javascript('galleries/expander_old')
+    else
+      posts_using = Post.where(icon_id: @icon.id)
+      replies_using = Reply.where(icon_id: @icon.id)
+      if logged_in?
+        unless @icon.user == current_user
+          posts_using = posts_using.where(privacy: [Concealable::PUBLIC, Concealable::REGISTERED])
+          replies_using = replies_using.where(posts: {privacy: [Concealable::PUBLIC, Concealable::REGISTERED]}).joins(:post)
```

Comment:
Can we put the `joins` before the `where`? Makes it easier for me to logic about it.